### PR TITLE
update to use OpenSplice 6.7

### PR DESCRIPTION
**Pull request by dirk-thomas**
This is the first step getting the ROS 2 code to compile against OpenSplice 6.7.

The implementation needs to be updated with all the recent changes (partitions, topic prefixes, etc.). So currently many tests are still failing. Also CI is not able to use OpenSplice 6.7 atm.

### CODE REVIEW

**Comment by dirk-thomas** [comment]
@mikaelarguedas I can address the deprecated option but likely in a separate PR. I would first like to get this into a state where master can be CI tested again.

This patch finally passes the build phase on Linux: [Build Status](http://ci.ros2.org/job/ci_linux/2879/)

It uses a custom Debian which will be replaced with a new "final" .deb in the process of being built and placed in packages.osrfoundation.org. Once that is available I will make a corresponding PR to the CI repo.

Therefore this PR is ready for review.

File: opensplice_cmake_module/cmake/Modules/FindOpenSplice.cmake [hunk 1]
```diff
@@ -76,7 +79,7 @@ if(NOT _ospl_home STREQUAL "")
     "ddsserialization"
     "ddsuser"
     "ddsutil"
-    "durability"
+    # "durability"
```
Comment (mikaelarguedas) [inline opensplice_cmake_module/cmake/Modules/FindOpenSplice.cmake]: does that library not exist at all ? or just not built properly in our custom deb ?
I think we should either put a comment here or remove the line depending on what makes most sense

File: rosidl_typesupport_opensplice_cpp/resource/msg__type_support.cpp.em [hunk 2]
```diff
@@ -28,6 +28,7 @@
 #include "rosidl_typesupport_opensplice_cpp/identifier.hpp"
 #include "rosidl_typesupport_opensplice_cpp/message_type_support.h"
 #include "rosidl_typesupport_opensplice_cpp/message_type_support_decl.hpp"
+#include "rosidl_typesupport_opensplice_cpp/u__instanceHandle.h"
```
Comment (mikaelarguedas) [inline rosidl_typesupport_opensplice_cpp/resource/msg__type_support.cpp.em]: is the semantic of this header different ? (to use the double underscore while every other header has a single one)

Comment (dirk-thomas) [inline rosidl_typesupport_opensplice_cpp/resource/msg__type_support.cpp.em]: The file is coming from OpenSplice. I didn't change their naming scheme.

File: opensplice_cmake_module/cmake/Modules/FindOpenSplice.cmake (hunk 1 above)
Comment (dirk-thomas) [inline opensplice_cmake_module/cmake/Modules/FindOpenSplice.cmake]: Good point. As far as I know it doesn't exist anymore. I will update the commit which commented it out to actually remove the line.

**Pull request merged**